### ISSUE TITLE
Need API to provide Client app details, given a client_id

### ISSUE THREAD

**Issue by kosso** [appdotnet/api-spec#242]
I have an urgent need for an API to be able to retrieve the details of a client app (the 'App Object'?) when given a client_id

This is for a service being developed where devs can register their apps to provide their apps and customers with extra specific functionality.  

(I've said too much) :)
## 

Possibly related: it would also be great to have an API to access and list the apps in the App.net App Directory categories. 
https://github.com/appdotnet/api-spec/issues/234

**Comment by berg** [comment]
Maybe just automate the delegate flow somehow? The only details you'd get are name/URL, and you could just retrieve those by having the user generate a delegate token for your app and verifying it.

**Comment by kosso** [comment]
Yes. After thinking more about this, I'm now using the delegate flow for
what I need in this particular case in the system I'm building.

And getting a dev to generate a delegate token is exactly what I'm asking
them to do now. :D

If they can't get that far, then the system I'm building (to allow a dev to
register an app with my service) will be pretty useless anyway ;)

**Comment by kosso** [comment]
More thoughts on this (since making some more progress on my system) :

It would be nice if the app data (name, client_id, link) supplied
by /stream/0/token also contained the user_id of the dev who created the
app.

(Since my system requires a developer to 'register' an app, it would be
useful to know if it was _their_ app - despite the possible lack of
usefulness to anyone else without a client secret if an app isn't theirs)

K
